python/ml4ir/model/pipeline.py

```diff
@@ -244,15 +244,34 @@ def run(self):
                 ExecutionModeKey.TRAIN_INFERENCE,
                 ExecutionModeKey.INFERENCE_EVALUATE,
                 ExecutionModeKey.INFERENCE_ONLY,
+                ExecutionModeKey.INFERENCE_EVALUATE_RESAVE,
+                ExecutionModeKey.INFERENCE_RESAVE,
             }:
                 # Predict ranking scores
-                model.predict(
+                ranking_model.predict(
                     test_dataset=ranking_dataset.test,
                     inference_signature=self.args.inference_signature,
                     logs_dir=self.logs_dir,
                     logging_frequency=self.args.logging_frequency,
                 )
 
+            # Save model
+            # NOTE: Model will be saved with the latest serving signatures
+            if self.args.execution_mode in {
+                ExecutionModeKey.TRAIN_INFERENCE_EVALUATE,
+                ExecutionModeKey.TRAIN_EVALUATE,
+                ExecutionModeKey.TRAIN_INFERENCE,
+                ExecutionModeKey.TRAIN_ONLY,
+                ExecutionModeKey.INFERENCE_EVALUATE_RESAVE,
+                ExecutionModeKey.EVALUATE_RESAVE,
+                ExecutionModeKey.INFERENCE_RESAVE,
+                ExecutionModeKey.RESAVE_ONLY,
+            }:
+                # Save model
+                ranking_model.save(
+                    models_dir=self.models_dir, pad_records=self.args.pad_records_at_inference
```

Comment:
so why do we have the `pad_records_at_inference` flag?  When would we want to change that flag's value?

Reply:
I'm just leaving an option open. Not padding limits us to only predict scores on one query at a time. If for some use case, we care more about batch inference and less about padded records, then we can use this.

`pad_records_at_inference` is set to False by default. So should be fine.